test/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDaoTest.java

```diff
@@ -90,7 +112,48 @@ public void canCountStudyParticipants() throws Exception {
         count = userConsentDao.getNumberOfParticipants(STUDY_IDENTIFIER);
         assertEquals("Correct number of participants", 5, count);
     }
+    
+    @Test
+    public void canGetAllConsentsInHistory() throws Exception {
+        final DynamoStudyConsent1 consent = createStudyConsent();
+        final StudyIdentifier studyId = new StudyIdentifierImpl(consent.getStudyKey());
+        for (int i=0; i < 5; i++) {
+            userConsentDao.giveConsent(HEALTH_CODE, consent, DateTime.now().getMillis());
+            userConsentDao.withdrawConsent(HEALTH_CODE, studyId, DateTime.now().getMillis());
+        }
+        userConsentDao.giveConsent(HEALTH_CODE, consent, DateTime.now().getMillis());
+        
+        List<UserConsent> consents = userConsentDao.getUserConsentHistory(HEALTH_CODE, studyId);
+        assertEquals(6, consents.size());
+        for (int i=0; i < 5; i++) {
+            assertNotNull(consents.get(i).getWithdrewOn());
+        }
+        assertNull(consents.get(consents.size()-1).getWithdrewOn());
+    }
 
+    @Test
+    public void getConsentBySignedOnDate() throws Exception {
```

Comment:
Can you add multiple consents with the same health code but different signedOn? This way, we can make sure the range key is being captured correctly by DDB.